### PR TITLE
dnsproxy: add more options

### PR DESCRIPTION
**Pull request by muink**
Maintainer: @1715173329 

add options:
- --http3
- --timeout

### CODE REVIEW

**Comment by 1715173329** [comment]
> add options:
>
> - --http3
> - --timeout

Please write this into your commit message as well. And it would be nice to have a simple description for them.

**Comment by muink** [comment]
Options description now added

**Review by 1715173329 — APPROVED** [review]

**Comment by BKPepe** [comment]
I have some doubts about this. It looks like adding some options to config file(s). Are you sure that does option does not require new dependencies?

**Comment by 1715173329** [comment]
This is a single static go program.

File: net/dnsproxy/files/dnsproxy.init [hunk 1]
```diff
@@ -52,6 +52,8 @@ load_config_arg() {
 }
 
 load_config_list() {
+	is_empty "global" "listen_addr" || config_list_foreach "global" "listen_addr" "append_param '--listen'"
+	is_empty "global" "listen_port" && append_param_arg "global" "listen_port" "--port" "5353" || config_list_foreach "global" "listen_port" "append_param '--port'"
```
Comment (1715173329) [inline net/dnsproxy/files/dnsproxy.init:56]: Please use "if else" statement here.

**Comment by 1715173329** [comment]
Please squash your commits into single one.

File: net/dnsproxy/Makefile [hunk 2]
```diff
@@ -6,7 +6,7 @@ include $(TOPDIR)/rules.mk
 
 PKG_NAME:=dnsproxy
 PKG_VERSION:=0.56.2
-PKG_RELEASE:=1
+PKG_RELEASE:=3
```
Comment (1715173329) [inline net/dnsproxy/Makefile:9]: This should be 2

File: net/dnsproxy/files/dnsproxy.init [hunk 3]
```diff
@@ -44,13 +44,20 @@ append_param_bool() {
...
 	append_param_bool "$1" "verbose"
 }
 
 load_config_list() {
+	is_empty "global" "listen_addr" || config_list_foreach "global" "listen_addr" "append_param '--listen'"
+	if $(is_empty "global" "listen_port"); then
+		append_param_arg "global" "listen_port" "--port" "5353"
+	else 
+		config_list_foreach "global" "listen_port" "append_param '--port'"
+	fi
```
Comment (1715173329) [inline net/dnsproxy/files/dnsproxy.init:60]:
```suggestion
	if is_empty "global" "listen_addr"; then
		append_param "--listen" "127.0.0.1"
	else 
		config_list_foreach "global" "listen_port" "append_param '--port'"
	fi

	if is_empty "global" "listen_port"; then
		append_param "--port" "5353"
	else 
		config_list_foreach "global" "listen_port" "append_param '--port'"
	fi
```

Addtionally, `config_list_foreach` won't work if the config type is `option`, so we need add a script to migrate old uci configuration.

We can do a simple string replacement:
```sh
#!/bin/sh

[ -s "/etc/config/dnsproxy" ] || exit 0

sed -i -e "s,option listen_addr,list listen_addr,g" -e "s,option listen_port,list listen_port,g" "/etc/config/dnsproxy"
exit 0
```

then put it into ` /etc/uci-defaults/80-dnsproxy-migration`

File: net/dnsproxy/Makefile [hunk 4]
```diff
@@ -49,6 +49,8 @@ define Package/dnsproxy/install
 	$(INSTALL_CONF) $(CURDIR)/files/dnsproxy.config $(1)/etc/config/dnsproxy
 	$(INSTALL_DIR) $(1)/etc/init.d/
 	$(INSTALL_BIN) $(CURDIR)/files/dnsproxy.init $(1)/etc/init.d/dnsproxy
+	$(INSTALL_DIR) $(1)/etc/uci-defaults
+	$(INSTALL_BIN) ./files/dnsproxy.defaults $(1)/etc/uci-defaults/80-dnsproxy-migration
```
Comment (1715173329) [inline net/dnsproxy/Makefile:53]:
```suggestion
	$(INSTALL_DIR) $(1)/etc/uci-defaults/
	$(INSTALL_BIN) $(CURDIR)/files/dnsproxy.defaults $(1)/etc/uci-defaults/80-dnsproxy-migration
```

Let's unify the style, and it's not necessary to split this migration commit.
Then I can merge it.

Comment (muink) [reply net/dnsproxy/Makefile:53]: All right

**Comment by 1715173329** [comment]
Merged, thank you!

**Pull request merged**